SabreTools.DatTools/Rebuilder.cs

```diff
@@ -239,8 +250,33 @@ public class Rebuilder
                         bool rebuilt = RebuildGenericHelper(datFile, file, outDir, quickScan, date, inverse, outputFormat, asFiles);
 
                         // If we are supposed to delete the file, do so
-                        if (delete && rebuilt)
-                            File.Delete(input);
+                        if (delete && rebuilt) 
+                        {
+                            try 
+                            {
+                                File.SetAttributes(file, FileAttributes.Normal);
+                                File.Delete(file);
+                                // sometimes the file is in a directory (TOSEC), delete directory
+                                Directory.Delete(Path.GetDirectoryName(file));
+                            }
+                            catch (Exception e)
+                            {
+                                logger.Warning("The process failed: " + e.Message);
+                            }
+                        }
+                    }
+                    
+                    // lopping through directories, delete them also 
```

Comment:
```suggestion
                    // Looping through directories, delete them also 
```

Additionally, I don't quite like using exceptions as the way to determine if the directory was empty or not. Throwing and catching exceptions is quite expensive and is not a good pattern to follow. I'd encourage something a little more safe that doesn't rely on that behavior.